embed-code-go/embedding/embedding_processor.go

```diff
@@ -0,0 +1,112 @@
+package embedding
+
+import (
+	"embed-code/embed-code-go/configuration"
+	"embed-code/embed-code-go/embedding/parsing"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// The EmbeddingProcessor entity processes a single documentation file and embeds code snippets
+// into it based on the provided configuration.
+//
+// DocFile — the path to the documentation file.
+//
+// Config — a configuration for embedding.
+type EmbeddingProcessor struct {
+	DocFile string
+	Config  configuration.Configuration
+}
+
+//
+// Initializers
+//
+
+// Creates and returns new EmbeddingProcessor with the given docFile and config.
+func NewEmbeddingProcessor(docFile string, config configuration.Configuration) EmbeddingProcessor {
+	return EmbeddingProcessor{
+		DocFile: docFile,
+		Config:  config,
+	}
+}
+
+//
+// Public methods
+//
+
+// Constructs embedding and modifys the doc file if embedding is needed.
+func (ep EmbeddingProcessor) Embed() {
+	context := ep.constructEmbedding()
+
+	if context.СheckContainsEmbedding() && context.СheckContentChanged() {
+		err := os.WriteFile(ep.DocFile, []byte(strings.Join(context.GetResult(), "\n")), 0644)
+		if err != nil {
+			panic(err)
+		}
+	}
+}
+
+// Reports whether the embedding of the target markdown is up-to-date with the code file.
+func (ep EmbeddingProcessor) CheckUpToDate() bool {
```

Comment:
I understand your point, but I think the main problem with the `check` method is that you do no expect `bool` as a result you expect more complicated responses with some `check` results, or even void type and some exceptions being thrown during the check. 
`Is` not always refer to getter-like functionality it could refer to some complex checks. The `return bool` type in the current context outweighs the complexity of the check by itself. Sofar I propose to switch to `is` or even rethink this somehow.